### ISSUE TITLE
Secrets has the AwsVersionId Key by default

### ISSUE THREAD

**Issue by james-nduka** [mumoshu/aws-secret-operator#42]
Can we avoid each secret being created to include the key for `AWSVersionID`?

We have a CRD that can only accept a single Key/Value and it is currently not compatible with AWSSecret CRD's

**Comment by mumoshu** [comment]
@james-nduka Hey! It's not configurable today, but technically possible. We only use AWSVersionID for change detection:

https://github.com/mumoshu/aws-secret-operator/search?q=AWSVersionID

We may instead use another store, like some annotation on the secret resource for the same purpose, which would help your use-case.

Probably it should be configurable via some AwsSecret field?

**Comment by ChangSurrey** [comment]
I agree that if it's only for change detection purpose, using an `annotation` makes more sense than including the version ID as part of the `AWSSecret` CRD itself.
